### PR TITLE
Fix subtle memory leak around AsyncTimer

### PR DESCRIPTION
**Pull request by Ignition**
--query-execution-timeout-sec causes an allocation which was small, but not being freed. This is now freed.

### CODE REVIEW

**Comment by Ignition** [comment]
### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

File: src/query/interpreter.cpp [hunk 1]
```diff
@@ -5493,7 +5494,7 @@ Interpreter::PrepareResult Interpreter::Prepare(const std::string &query_string,
   // Explicit transactions define the metadata at the beginning and reuse it
   spdlog::debug(
       "{}", QueryLogWrapper{query_string,
-                            (in_explicit_transaction_ && metadata_ && !is_begin) ? &*metadata_ : &extras.metadata_pv,
+                            (active_transaction_ && !is_begin) ? &active_transaction_->metadata_ : &extras.metadata_pv,
```
Comment (andrejtonev) [inline src/query/interpreter.cpp:5497]: I think this should be active and active->explicit

File: src/glue/SessionHL.cpp [hunk 2]
```diff
@@ -395,7 +395,8 @@ void SessionHL::Configure(const bolt_map_t &run_time_info) {
     }
     in_explicit_db_ = true;
     // NOTE: Once in a transaction, the drivers stop explicitly sending the db and count on using it until commit
-  } else if (in_explicit_db_ && !interpreter_.in_explicit_transaction_) {  // Just on a switch
+  } else if (in_explicit_db_ && interpreter_.active_transaction_ &&
```
[the remark this answers is not in the document]
Comment (Ignition) [reply src/glue/SessionHL.cpp:398]: agree

File: src/query/interpreter.cpp (hunk 1 above)
Comment (Ignition) [reply src/query/interpreter.cpp:5497]: agree

**Review by andrejtonev — APPROVED** [review]

**Pull request merged**